tests/bundle/RequestParser/RouterTest.php

```diff
@@ -83,77 +82,89 @@ public function testParseHref()
+        $this->expectExceptionMessage(
+            'No attribute \'badAttribute\' in route matched from /api/test/v1/content/no-attribute'
+        );
 
         $href = '/api/test/v1/content/no-attribute';
 
         $matchResult = [
             '_route' => 'ibexa.rest.test_parse_href_attribute_not_found',
         ];
 
-        $this->getRouterMock()
-            ->expects($this->once())
-            ->method('matchRequest')
-            ->willReturn($matchResult);
+        $this->getRouterInvocationMockerForMatchingUri($href)
+             ->willReturn($matchResult)
+        ;
 
         self::assertEquals(1, $this->getRequestParser()->parseHref($href, 'badAttribute'));
     }
 
-    public function testGenerate()
+    public function testGenerate(): void
     {
         $routeName = 'ibexa.rest.test_generate';
         $arguments = ['arg1' => 1];
 
         $expectedResult = self::$routePrefix . '/generate/' . $arguments['arg1'];
         $this->getRouterMock()
-            ->expects($this->once())
-            ->method('generate')
-            ->with($routeName, $arguments)
-            ->willReturn($expectedResult);
+             ->expects($this->once())
+             ->method('generate')
+             ->with($routeName, $arguments)
+             ->willReturn($expectedResult)
+        ;
 
         self::assertEquals(
             $expectedResult,
             $this->getRequestParser()->generate($routeName, $arguments)
         );
     }
 
-    /**
-     * @return \Ibexa\Bundle\Rest\RequestParser\Router
-     */
-    private function getRequestParser()
+    private function getRequestParser(): RequestParser
     {
+        $routerMock = $this->getRouterMock();
+
         return new RouterRequestParser(
-            $this->getRouterMock()
+            $routerMock,
+            new UriParser($routerMock)
         );
     }
 
     /**
-     * @return \Symfony\Cmf\Component\Routing\ChainRouter|\PHPUnit\Framework\MockObject\MockObject
+     * @return \Symfony\Component\Routing\RouterInterface&\PHPUnit\Framework\MockObject\MockObject
      */
-    private function getRouterMock()
+    private function getRouterMock(): RouterInterface
     {
         if (!isset($this->router)) {
-            $this->router = $this->createMock(ChainRouter::class);
+            $this->router = $this->createMock(RouterInterface::class);
 
             $this->router
-                ->expects($this->any())
                 ->method('getContext')
-                ->willReturn(new RequestContext());
+                ->willReturn(new RequestContext())
+            ;
         }
 
         return $this->router;
     }
+
+    private function getRouterInvocationMockerForMatchingUri(
+        string $uri,
+        ?InvokedCountMatcher $invokedCount = null
+    ): InvocationMocker {
+        return $this->getRouterMock()
+                    ->expects($invokedCount ?? self::once())
+                    ->method('match')
+                    ->with($uri)
```

Comment:
I think indentation is wrong here.

Reply:
> I think indentation is wrong here.

@konradoboza this is how PHPStorm auto-formats it for me. It's aligned with the first `->` in L162 for readability. What indentation would work better for you?

Reply:
I'm more used to sth like:
```
return $this->getRouterMock()
    ->expects($invokedCount ?? self::once())
    ->method('match')
    ->with($uri)
```
but this is only a nitpick, feel free to leave it as-is, I approved the PR regardless.

Reply:
No strong opinion from my side, so fixed via 4ef4eaa :)

Reply:
Thank you. 💪